include/zephyr/net/ethernet.h

```diff
@@ -452,6 +468,9 @@ struct ethernet_api {
 
 	/** Send a network packet */
 	int (*send)(const struct device *dev, struct net_pkt *pkt);
+
+	/** Types of Ethernet L2 network interfaces */
+	enum ethernet_types l2_eth_type;
```

Comment:
The reason I have chosen the API is because drivers can easily register the type, putting them in context makes it a tough. 

> Plus, I'm not really convinced we need new enum ethernet_types type - IMO it should be implicit we deal with a Ethernet device, maybe just add bool is_wifi : 1, compiled in only if Wi-Fi is enabled could be added instead?

True, enum is just there for future proofing, esp. I am working on another PR that introduces WIFI_NATIVE, that can be used to handle WPA supplicant (downstream, there is no native Wi-Fi in upstream).

Reply:
> > The reason I have chosen the API is because drivers can easily register the type, putting them in context makes it a tough.
> 
> That's why I propose to make Ethernet implicit, so that only Wi-Fi devices would need to set the custom flag.

That is what I am doing using default `0` for enum (though a bit convoluted).

> > True, enum is just there for future proofing, esp. I am working on another PR that introduces WIFI_NATIVE, that can be used to handle WPA supplicant (downstream, there is no native Wi-Fi in upstream).
> 
> But is it really different from the device driver perspective, whether the upper layer (supplicant) is offloaded or not?

Well, only driver knows what kind of Wi-Fi functionality is offloaded, so, it gives a hint to the stack and then we can do something like below in Wi-Fi management:

```
if (net_if_wifi_native(iface)) {
   iface->native_api->connect()
else
   off_api->connect()
}
```
And we can have per-iface implementations of Native (I am still thinking about tying native with a iface or keeping it iface agnostic).

Reply:
But such aspects (IMHO) whether to use native or offloaded Wi-Fi management with the specific interface should not be buried inside Ethernet L2, there's a `struct net_wifi_mgmt_offload` (which should probably be renamed at that into something more generic) which could carry whatever Wi-Fi specific configuration you need (likely by grouping native/offloaded APIs into separate structs and setting them accordingly per interface).

I'm also not sure - should native/offloaded Wi-Fi mgmt APIs really differ? IMO it'd be cleaner to have a single mgmt interface (based on the current one) and simply implement it for the native solution. Offloaded interfaces could come up with their own implementations as they do now.